src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java

```diff
@@ -163,5 +166,16 @@ private SelfDescribingJson getFinalPost(List<TrackerPayload> buffer) {
     @Override
     public void close() {
         flushBuffer();
+        if (executor != null) {
+            executor.shutdown();
+            try {
+                boolean e = executor.awaitTermination(closeTimeout, TimeUnit.SECONDS);
+                if (e == false) {
+                    LOGGER.debug("Executor service shut down after time out");
+                }
```

Comment:
if `e` is false, it means that the executor wasn't shutdown.

If you want some inspiration you can have a look at https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html